lib/services/history/history_manager.ts

```diff
@@ -147,12 +147,12 @@ export class SNHistoryManager extends PureService {
     if (entry && this.persistable) {
       /** Debounce, clear existing timeout */
       if (this.saveTimeout) {
-        if (this.timeout.hasOwnProperty('cancel')) {
+        if (Object.prototype.hasOwnProperty.call(this.timeout, 'cancel')) {
```

Comment:
I assume a linter error triggered this change? This is much less straightforward to read, there has to be a way to keep the existing syntax, maybe in the type signature?

Reply:
Yes, this was caused by this: https://eslint.org/docs/rules/no-prototype-builtins

used the solution suggested in examples. I'm open to suggestions on how to change this

Reply:
I guess the underlying issue is using `hasOwnProperty`. And checking the prototype chain is something we should do in this case, so what do you think of this:
```js
if ('cancel' in this.timeout) {
  // ...
}
```